### PR TITLE
Fix hook execution order on non-nullable field error

### PR DESCRIPTION
**Pull request by user1736**

### CODE REVIEW

File: packages/supermassive/src/executeWithoutSchema.ts [hunk 1]
```diff
@@ -1061,6 +1073,22 @@ function resolveAndCompleteField(
   } catch (rawError) {
     const pathArray = pathToArray(path);
     const error = locatedError(rawError, fieldGroup, pathArray);
+
+    const fieldKey = pathToString(info.path);
+    const hookContextMap = exeContext.hookContextMap;
+    for (const [hookKey, hookData] of hookContextMap.entries()) {
+      if (hookKey.startsWith(fieldKey) && hookKey !== fieldKey) {
+        // On error call complete hooks for all the nested fields, that might otherwise be called too late
+        invokeAfterFieldCompleteHook(
```
Comment (user1736) [inline packages/supermassive/src/executeWithoutSchema.ts:1082]: WIP just do demonstrate the idea. Will add call for `afterFieldResolved` hook later

Comment (user1736) [reply packages/supermassive/src/executeWithoutSchema.ts:954]: done

File: packages/supermassive/src/executeWithoutSchema.ts [hunk 2]
```diff
@@ -1061,6 +1073,22 @@ function resolveAndCompleteField(
...
+          hookData.resolveInfo,
+          exeContext,
+          undefined, // FIXME not used
+          undefined,
+          error,
```
[the remark this answers is not in the document]
Comment (user1736) [reply packages/supermassive/src/executeWithoutSchema.ts:958]: done

File: packages/supermassive/src/__tests__/hooks.test.ts [hunk 3]
```diff
@@ -371,6 +371,44 @@ describe.each([
...
+          "AFR|person|[object]|undefined",
+          "BFR|person.name",
+          "BFR|person.xField",
+          "AFR|person.xField|undefined|Resolver error",
+          "AFC|person.xField|undefined|Resolver error",
+          'AFR|person.name|undefined|Sibling non-nullable field threw: "Resolver error"',
+          'AFC|person.name|undefined|Sibling non-nullable field threw: "Resolver error"',
+          "AFC|person|undefined|Resolver error",
+          "ABR|GetPersonX",
+        ],
```
Comment (vladar) [inline packages/supermassive/src/__tests__/hooks.test.ts:408]: Could you post current behavior (i.e. before the PR?)

Comment (user1736) [reply packages/supermassive/src/__tests__/hooks.test.ts:408]: @vladar it was like this:

```
"BOE|GetPersonX",
"BFR|person",
"AFR|person|[object]|undefined",
"BFR|person.name",
"BFR|person.xField",
"AFR|person.xField|undefined|Resolver error",
"AFC|person.xField|undefined|Resolver error",
"AFC|person|undefined|Resolver error",
"ABR|GetPersonX",
"AFR|person.name|John Doe|undefined",
```

**Comment by user1736** [comment]
abandoning this PR in favour of #492 that adops solution from `graphql-js`

**Comment by user1736** [comment]
abandoning this PR in favour of #492 that adopts solution from `graphql-js`

**Pull request closed without merging**